git_machete/cmd.py

```diff
@@ -1094,20 +1102,83 @@ def merge_base(cli_ctxt: CommandLineContext, sha1: str, sha2: str) -> str:
 
 # Note: the 'git rev-parse --verify' validation is not performed in case for either of earlier/later
 # if the corresponding prefix is empty AND the revision is a 40 hex digit hash.
-def is_ancestor(cli_ctxt: CommandLineContext, earlier_revision: str, later_revision: str, earlier_prefix: str = "refs/heads/", later_prefix: str = "refs/heads/") -> bool:
-    if earlier_prefix == "" and is_full_sha(earlier_revision):
-        earlier_sha: Optional[str] = earlier_revision
-    else:
-        earlier_sha = commit_sha_by_revision(cli_ctxt, earlier_revision, earlier_prefix)
-    if later_prefix == "" and is_full_sha(later_revision):
-        later_sha: Optional[str] = later_revision
-    else:
-        later_sha = commit_sha_by_revision(cli_ctxt, later_revision, later_prefix)
+def is_ancestor(
+    cli_ctxt: CommandLineContext,
+    earlier_revision: str,
+    later_revision: str,
+    earlier_prefix: str = "refs/heads/",
+    later_prefix: str = "refs/heads/",
+) -> bool:
+    earlier_sha = full_sha(cli_ctxt, earlier_revision, earlier_prefix)
+    later_sha = full_sha(cli_ctxt, later_revision, later_prefix)
+
     if earlier_sha == later_sha:
         return True
     return merge_base(cli_ctxt, earlier_sha, later_sha) == earlier_sha
 
 
+# Determine if later_revision, or any ancestors of later_revision, contain a
```

Comment:
Nit, for clarify:
```suggestion
# Determine if later_revision, or any ancestors of later_revision that are NOT ancestors of earlier_revision, contain a
```

Since just _any ancestors of later_revision_ would be usually a looot of commits to check ;)